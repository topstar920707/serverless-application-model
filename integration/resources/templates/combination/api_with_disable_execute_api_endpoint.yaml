Parameters:
  DisableExecuteApiEndpointValue:
    Description: Variable to define if client can access default API endpoint.
    Type: String
    AllowedValues: [true, false]

Resources:
  RestApiGateway:
    Type: AWS::Serverless::Api
    Properties:
      StageName: Prod
      DisableExecuteApiEndpoint:
        Ref: DisableExecuteApiEndpointValue

  RestApiFunction:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: |
        exports.handler = async (event) => {
          const response = {
            statusCode: 200,
            body: JSON.stringify('Hello from Lambda!'),
          };
          return response;
        };
      Handler: index.handler
      Runtime: nodejs14.x
      Events:
        Iam:
          Type: Api
          Properties:
            RestApiId: !Ref RestApiGateway
            Method: GET
            Path: /
Outputs:
  ApiUrl:
    Description: "API endpoint URL for Prod environment"
    Value:
      Fn::Sub: 'https://${RestApiGateway}.execute-api.${AWS::Region}.${AWS::URLSuffix}/Prod/'