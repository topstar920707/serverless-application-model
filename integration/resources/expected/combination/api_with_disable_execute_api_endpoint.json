[
    {"LogicalResourceId": "RestApiGateway", "ResourceType": "AWS::ApiGateway::RestApi"},
    {"LogicalResourceId": "RestApiGatewayDeployment", "ResourceType": "AWS::ApiGateway::Deployment"},
    {"LogicalResourceId": "RestApiGatewayProdStage", "ResourceType": "AWS::ApiGateway::Stage"},
    {"LogicalResourceId": "RestApiFunction", "ResourceType": "AWS::Lambda::Function"},
    {"LogicalResourceId": "RestApiFunctionIamPermissionProd", "ResourceType": "AWS::Lambda::Permission"},
    {"LogicalResourceId": "RestApiFunctionRole", "ResourceType": "AWS::IAM::Role"}
]
